rules/library.bzl

```diff
@@ -538,39 +594,31 @@ def apple_library(name, library_tools = {}, export_private_headers = True, names
             name = import_name + "_vfs",
             framework_name = vfs_framework_name,
             modulemap = import_module_map,
+            swiftmodule_contents = import_swiftmodules,
             hdrs = import_headers,
             tags = _MANUAL,
         )
         import_vfsoverlays.append(import_name + "_vfs")
 
-    for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
-        fail("no import for %s" % vendored_dynamic_library)
-        import_headers += native.glob(
+    for xcframework in kwargs.pop("vendored_xcframeworks", []):
+        import_headers = native.glob(
             ["**/*.h"],
         )
-        import_module_map = native.glob(
+        import_module_maps = native.glob(
             ["**/*.modulemap"],
         )
-
         if len(import_module_maps) > 0:
             import_module_map = import_module_maps[0]
         else:
             import_module_map = None
-
         vfs_imported_framework = _find_imported_framework_name(import_headers)
-        vfs_framework_name = vfs_imported_framework if vfs_imported_framework else namespace
-        framework_vfs_overlay(
-            name = import_name + "_vfs",
-            framework_name = vfs_framework_name,
-            modulemap = import_module_map,
-            hdrs = import_headers,
-            tags = _MANUAL,
-        )
-        import_vfsoverlays.append(import_name + "_vfs")
+        xcframework_name, xcframework_name_vfs = _xcframework(library_name = name, **xcframework)
+        vendored_deps.append(xcframework_name)
+        import_vfsoverlays.append(xcframework_name_vfs)
 
-    for xcframework in kwargs.pop("vendored_xcframeworks", []):
-        vendored_deps.append(_xcframework(library_name = name, **xcframework))
-        # TODO: for virtual frameworks we'll need to add this
+    # We don't currently support dynamic library
+    for vendored_dynamic_library in kwargs.pop("vendored_dynamic_libraries", []):
+        fail("no support for dynamic library: %s" % vendored_dynamic_library)
```

Comment:
This always fails in the first iteration right? Do you need a for loop for this?